app/views/posts/index.html.erb

```diff
@@ -0,0 +1,20 @@
+<h1>Posts</h1>
+<% @posts.each do |post| %>
+    <div class="card">
+      <div class="card-header font-weight-bold h4">
+        Post
+      </div>
+      <% if user_signed_in? %>
+        <div class="card-body">
+          <p class="card-text lead"><%= post.content %></p>
+          <hr class="my-4">
+          <h6 class="text-right card-title font-weight-bold">Posted by: <%= post.user_email.split('@').first.capitalize %></h6>
+        </div>
+      <% else%>
+        <div class="card-body">
+          <p class="card-text lead"><%= post.content %></p>
+        </div>
+      <% end %>
```

Comment:
Kindly move all view-related logic to the view helper.
There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.
      **Resources**
- [Where Do I Put My Code?](http://codefol.io/posts/Where-Do-I-Put-My-Code)
- [The Beginner's Guide to Rails Helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)
- [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright